tests/unit/test_grpc_index.py

```diff
@@ -0,0 +1,239 @@
+import pinecone
+from core.utils import dict_to_proto_struct
+from pinecone import DescribeIndexStatsRequest
+from pinecone.core.grpc.protos.vector_service_pb2 import Vector, DescribeIndexStatsRequest, UpdateRequest, \
+    UpsertRequest, FetchRequest, QueryRequest, DeleteRequest, QueryVector
+
+
+class TestRestIndex:
+
+    def setup_method(self):
```

Comment:
Side note: we should probably use `UnitTest.TestCase` class.  
Let's defer this to a task in the future \ next versions.